### ISSUE TITLE
Cannot retrieve SonarQube.Integration.targets

### ISSUE THREAD

**Issue by Jonathan34** [SonarOpenCommunity/sonar-cxx-msbuild-tasks#53]
Hi,

I gave the msbuild taks a try but it seems there is something wrong when it tires to find some files.
This is a log:

> Pre-processing started.
> Using environment variables to determine the download directory...
> Using environment variable 'AGENT_BUILDDIRECTORY', value 'C:\TFS\Builds\Agent404_01\1'
> Preparing working directories...
> Creating directory: C:\TFS\Builds\Agent404_01\1.sonarqube
> Creating directory: C:\TFS\Builds\Agent404_01\1.sonarqube\bin
> SonarQube server url: http://server-url:8888/
> Checking for updates...
> Downloading SonarQube.MSBuild.Runner.Implementation.zip from http://server-url:8888/static/csharp/SonarQube.MSBuild.Runner.Implementation.zip to C:\TFS\Builds\Agent404_01\1.sonarqube\bin\SonarQube.MSBuild.Runner.Implementation.zip
> Extracting files to C:\TFS\Builds\Agent404_01\1.sonarqube\bin...
> Executing file C:\TFS\Builds\Agent404_01\1.sonarqube\bin\MSBuild.SonarQube.Internal.PreProcess.exe
>   Args: /d:sonar.host.url=http://server-url:8888/ /k:PROJECTKEY /n:PROJECTNAME /v:1.0 /d:sonar.verbose=true /s:C:\Temp\tmpF878.tmp <sensitive data removed>
>   Working directory: C:\TFS\Builds\Agent404_01\1.sonarqube
>   Timeout (ms):-1
>   Process id: 7792
> SonarQube Scanner for MSBuild Begin Step 2.0
> 15:55:30.603  Loading analysis properties from C:\Temp\tmpF878.tmp
> 15:55:30.665  sonar.verbose=true was specified - setting the log verbosity to 'Debug'
> 15:55:30.681  Updating build integration targets...
> 15:55:30.681  The file SonarQube.Integration.ImportBefore.targets is up to date at C:\Users\AgentBuilder\AppData\Local\Microsoft\MSBuild\14.0\Microsoft.Common.targets\ImportBefore
> 15:55:30.681  The file SonarQube.Integration.ImportBefore.targets is up to date at C:\Users\AgentBuilder\AppData\Local\Microsoft\MSBuild\12.0\Microsoft.Common.targets\ImportBefore
> 15:55:30.681  Creating config and output folders...
> 15:55:30.681  Creating directory: C:\TFS\Builds\Agent404_01\1.sonarqube\conf
> 15:55:30.697  Creating directory: C:\TFS\Builds\Agent404_01\1.sonarqube\out
> 15:55:30.712  Fetching analysis configuration settings...
> 15:55:30.728  Fetching properties for project 'PROJECTKEY' from http://server-url:8888/api/properties?resource=PROJECTKEY...
> 15:55:30.728  Downloading from http://server-url:8888/api/properties?resource=PROJECTKEY...
> 15:55:31.744  Generating rulesets...
> 15:55:31.744  Generating the FxCop ruleset: C:\TFS\Builds\Agent404_01\1.sonarqube\conf\SonarQubeFxCop-cs.ruleset
> 15:55:31.744  Downloading from http://server-url:8888/api/updatecenter/installed_plugins...
> 15:55:32.403  Downloading from http://server-url:8888/api/profiles/list?language=cs&project=PROJECTKEY...
> 15:55:32.981  Downloading from http://server-url:8888/api/profiles/index?language=cs&name=Sonar+way...
> 15:55:33.528  Generating the FxCop ruleset: C:\TFS\Builds\Agent404_01\1.sonarqube\conf\SonarQubeFxCop-vbnet.ruleset
> 15:55:33.528  Downloading from http://server-url:8888/api/updatecenter/installed_plugins...
> 15:55:34.09  Downloading from http://server-url:8888/api/updatecenter/installed_plugins...
> 15:55:34.684  Downloading from http://server-url:8888/api/profiles/list?language=cs&project=PROJECTKEY...
> 15:55:35.17  Downloading from http://server-url:8888/profiles/export?format=roslyn-cs&language=cs&name=Sonar+way...
> 15:55:36.202  Found profile export 'roslyn-cs' for project 'PROJECTKEY'
> 15:55:36.265  Writing Roslyn generated ruleset to C:\TFS\Builds\Agent404_01\1.sonarqube\conf\SonarQubeRoslyn-cs.ruleset...
> 15:55:36.281  Writing Roslyn analyzer additional file to C:\TFS\Builds\Agent404_01\1.sonarqube\conf\SonarLint.xml...
> 15:55:36.281  Provisioning analyzer assemblies...
> 15:55:36.296  Installing required Roslyn analyzers...
> 15:55:36.296  Local analyzer cache: C:\Temp.sonarqube.static
> 15:55:36.296  Processing plugin: csharp version 1.10.0
> 15:55:36.296  Cache hit: using plugin files from C:\Temp.sonarqube.static\csharp_1.10.0\SonarAnalyzer.zip
> Process returned exit code 0
> Pre-processing succeeded.
> Exception During Run: Could not find a part of the path 'C:\TFS\Builds\Agent404_01\1\s\Acidizing.sonarqube\bin\Targets\SonarQube.Integration.targets'. 
>     at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
>    at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
>    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
>    at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize, Boolean checkHost)
>    at System.IO.StreamReader..ctor(String path, Encoding encoding)
>    at System.IO.File.InternalReadAllLines(String path, Encoding encoding)
>    at Options.PatchMSbuildSonarRunnerTargetsFiles(String targetFile, OptionsData options) in C:\projects\sonar-cxx-msbuild-tasks\CxxSonarQubeMsbuidRunner\Options.fs:line 709
>    at Program.main(String[] argv) in C:\projects\sonar-cxx-msbuild-tasks\CxxSonarQubeMsbuidRunner\Program.fs:line 43

It is trying to find 'C:\TFS\Builds\Agent404_01\1\s\Acidizing.sonarqube\bin\Targets\SonarQube.Integration.targets' at the wrong place: it should be 'C:\TFS\Builds\Agent404_01\1.sonarqube\bin\Targets\SonarQube.Integration.targets'. 

I ve tried to set the working dir in parameter to the runner, but it does not work.
the /m parameter of the runner is /m:Acidizing\MySLN.sln, but i do nothave much choice i guess as this sln includes modules from the parent folder.

**Comment by jmecosta** [comment]
Hi

Can you try with absolute path to solution?

**Comment by Jonathan34** [comment]
Same issue with absolute path

**Comment by jmecosta** [comment]
seems it seems that it uses this AGENT_BUILDDIRECTORY to drop the target files, i will prepare something for u to try

**Comment by jmecosta** [comment]
check if this does it

https://github.com/jmecsoftware/sonar-cxx-msbuild-tasks/releases/tag/1.9.5-Beta1

**Comment by Jonathan34** [comment]
wish there was a way to chat with you instead :)
it still does the same:

> Using environment variable 'AGENT_BUILDDIRECTORY', value 'C:\TFS\Builds\Agent404_01\1'
> Exception During Run: Could not find a part of the path 'C:\TFS\Builds\Agent404_01\1\s\Acidizing.sonarqube\bin\Targets\SonarQube.Integration.targets'. 
>     at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
>    at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
>    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
>    at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize, Boolean checkHost)
>    at System.IO.StreamReader..ctor(String path, Encoding encoding)
>    at System.IO.File.InternalReadAllLines(String path, Encoding encoding)
>    at Options.PatchMSbuildSonarRunnerTargetsFiles(String targetFile, OptionsData options) in C:\projects\sonar-cxx-msbuild-tasks\CxxSonarQubeMsbuidRunner\Options.fs:line 709
>    at Program.main(String[] argv) in C:\projects\sonar-cxx-msbuild-tasks\CxxSonarQubeMsbuidRunner\Program.fs:line 43

can you output the version of the runner somewhere in the log to ensure it uses the right runner (i tried twice so i m confident it s updated)

**Comment by jmecosta** [comment]
looks like you are not using the last version. line 709 is not there anymore https://github.com/jmecsoftware/sonar-cxx-msbuild-tasks/blob/master/CxxSonarQubeMsbuidRunner/Options.fs#L709 

ive redone the zip in the same release. try unziping and deleting the old version

**Comment by Jonathan34** [comment]
Ok, tried again and it passed. thanks!
but now it fails with errors like:

>  C:\TFS\Builds\Agent404_01\1\s\Common\Model.Test\Model.Test.vcxproj(23,3): error MSB4019: The imported project "C:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk. 

Seems like $(VCTargetsPath) is empty...? it builds with tools version 14.0
i ll investigate but it seems to only happen with that runner (did not have that error before)

**Comment by jmecosta** [comment]
Seems your c++ compiler environment is bit messed. Are you able to compile
outside the wrapper. Perhaps combination of configuration and platform?

**Comment by Jonathan34** [comment]
yes everything works fine if i open the sln and clean/build or even if i build using TFS default steps (without the cxx runner).
If i manually invoke the cxxrunner, it fails.

The following works: 

> "C:\Program Files (x86)\MSBuild\12.0\Bin\MSBuild.exe" "C:\TFS\Builds\Agent404_01\1\s\Acidizing\CoralPlugin.sln" /v:Detailed  /t:Clea
> n;Build /m:1 /p:VisualStudioVersion=1.0

but with "C:\Program Files (x86)\MSBuild\14.0\Bin\MSBuild.exe", it does not...
i will try to uninstall and reinstall the microsft build tools. that's odd!

**Comment by Jonathan34** [comment]
For some reasons, it keeps downloading the 14.0 integration target but i have uninstalled the msbuild 14.0 and rebooted.

> 14:45:14.486  Updating build integration targets...
> 14:45:14.486  The file SonarQube.Integration.ImportBefore.targets is up to date at C:\Users\CoralBuild\AppData\Local\Microsoft\MSBuild\14.0\Microsoft.Common.targets\ImportBefore
> 14:45:14.486  The file SonarQube.Integration.ImportBefore.targets is up to date at C:\Users\CoralBuild\AppData\Local\Microsoft\MSBuild\12.0\Microsoft.Common.targets\ImportBefore

and then of course it fails caliing this as it does not exist "C:\Program Files\MSBuild\14.0\Bin\MSBuild.exe" 

any idea what you are using to invoke msbuild?

i see the script calls 

> Capture Environment For Build cmd.exe /c "C:\Program Files\Microsoft Visual Studio 12.0\VC\vcvarsall.bat " && set 

but vcvarsall.bat is under program files (x86) and this is specified in the environment variables...

**Comment by jmecosta** [comment]
right, default is vs 2015

try this 

```
        Console.WriteLine ("    /X|/x:<version of msbuild : vs10, vs12, vs13, vs15, default is vs15>")
```

**Comment by Jonathan34** [comment]
It displays this 

> The following msbuild processes have been found: 
> but i see the machine is building and the buildlog.txt is growing...so i ll just wait i guess :)

is it mandatory to build 1 project at a time? that is going to take age :(

**Comment by jmecosta** [comment]
Nop, the wrapper takes track of what msbuild processes are running before
it starts so I want killed them when it ends.  How many agents have you
running in same machine?

**Comment by Jonathan34** [comment]
On this machine I have 1 agent for the moment, but when this will be configured correctly, this build config will be run on 2 machines, one of them has 3 agents, this one will have 2 likely.
So it is likely the machine (any) will be doing something else in the meanwhile and be runnign another msbuild process.

**Comment by jmecosta** [comment]
We use this in machines with more than one agent, so far haven't spot
issues. However I think there is a option to stop the killing.

**Comment by Jonathan34** [comment]
Ok the build finished with error:

it starts with message like this:

>  The following msbuild processes have been found: 
>  The following msbuild processes have been found after running analysis: 
>  The following msbuild processes have been found after cleaning up: 
>                 AcidizingCatalogPackage.cpp will be compiled as it was not found in the tracking log. 
>                 AcidizingCoiledTubing.cpp will be compiled as it was not found in the tracking log. 

then later on:

> C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\x86_amd64\CL.exe /c /IC:\TFS\Builds\Agent404_01\1\s\Acidizing..\Common\Utility /IC:\TFS\Builds\Agent404_01\1\s\Acidizing..\Common\Attributes /IC:\TFS\Builds\Agent404_01\1\s\Acidizing..\Common\Catalogs /IC:\TFS\Builds\Agent404_01\1\s\Acidizing..\Common\Model /IC:\TFS\Builds\Agent404_01\1\s\Acidizing..\3rdParty\VLD\include /Zi /nologo /W4 /WX /sdl /Od /D CODE_ANALYSIS /D ACIDIZING_MODEL_EXPORTS /D _WINDOWS /D NOMINMAX /D UNICODE /D WIN32_LEAN_AND_MEAN /D WIN32 /D WIN64 /D _DEBUG /Gm /EHsc /RTC1 /MDd /GS /Gy- /fp:precise /Zc:wchar_t /Zc:forScope /openmp- /Yu"precompiled.h" /Fp"C:\TFS\Builds\Agent404_01\1\s\Acidizing\IntermediateFiles\Slb.Production.Acidizing.Model\x64\Debug\Slb.Production.Acidizing.ModelD.pch.codeanalysis" /Fo"C:\TFS\Builds\Agent404_01\1\s\Acidizing\IntermediateFiles\Slb.Production.Acidizing.Model\x64\Debug\" /Fd"C:\TFS\Builds\Agent404_01\1\s\Acidizing\IntermediateFiles\Slb.Production.Acidizing.Model\x64\Debug...
>                 Tracking command: 
>                 C:\Program Files (x86)\MSBuild\12.0\bin\Tracker.exe /d "C:\Program Files (x86)\MSBuild\12.0\bin\FileTracker.dll" /i C:\TFS\Builds\Agent404_01\1\s\Acidizing\IntermediateFiles\Slb.Production.Acidizing.Model\x64\Debug\Slb.Prod.12B7BBB1.tlog /r "C:\TFS\BUILDS\AGENT404_01\1\S\ACIDIZING\MODEL\ACIDEFFICIENCY.CPP|C:\TFS\BUILDS\AGENT404_01\1\S\ACIDIZING\MODEL\ACIDIZINGCATALOGPACKAGE.CPP|C:\TFS\BUILDS\AGENT404_01\1\S\ACIDIZING\MODEL\ACIDIZINGCOILEDTUBING.CPP|C:\TFS\BUILDS\AGENT404_01\1\S\ACIDIZING\MODEL\ACIDIZINGCOMPLETION.CPP|C:\TFS\BUILDS\AGENT404_01\1\S\ACIDIZING\MODEL\ACIDIZINGPERFORATION.CPP|C:\TFS\BUILDS\AGENT404_01\1\S\ACIDIZING\MODEL\ACIDIZINGPERFORATIONCONTAINER.CPP|C:\TFS\BUILDS\AGENT404_01\1\S\ACIDIZING\MODEL\ACIDIZINGSIMULATIONATTRIBUTES.CPP|C:\TFS\BUILDS\AGENT404_01\1\S\ACIDIZING\MODEL\ACIDIZINGSTUDY.CPP|C:\TFS\BUILDS\AGENT404_01\1\S\ACIDIZING\MODEL\ACIDIZINGZONATION.CPP|C:\TFS\BUILDS\AGENT404_01\1\S\ACIDIZING\MODEL\ACIDIZINGZONATIONFROMCOMPLETION.CPP|C:\TFS\BUILDS\AGENT404_01\1\S\ACIDIZ...
>                 IAcidizingSimulation.cpp 

and finally boom:

>  Running Code Analysis for C/C++... 
>  Failed to build project, check log in .cxxresults\BuildLog.txt
> 
> ---
> 
> Finishing task: CmdLine
> 
> ---
> 
> Task CmdLine failed. This caused the job to fail. Look at the logs for the task for more details.

The last line of the log file is the "running code analysis".

**Comment by jmecosta** [comment]
Check the buildlog.txt in Cxxresults the message in log is just the last
100 lines.  It should give hints what's going on.

The build is run with code analysis on, so there might be different things
going on that you don't expect.

Perhaps precompiled headers there is something in the log you sent.

**Comment by jmecosta** [comment]
@Jonathan34 you can try also to disable code analysis. since it was were it stopped

see
https://github.com/jmecsoftware/sonar-cxx-msbuild-tasks/blob/master/CxxSonarQubeMsbuidRunner/Options.fs#L160

it should be faster to build also

**Comment by Jonathan34** [comment]
Hi,
Thanks for your support. The build passed, everything is successful except i do not have any result in sonar.

*_1. *_Looking at the log:

```
 Attempting to locate the CodeCoverage.exe tool...
Code coverage command line tool: C:\Program Files (x86)\Microsoft Visual Studio 12.0\Team Tools\Dynamic Code Coverage Tools\CodeCoverage.exe
Fetching code coverage report information from TFS...
Attempting to locate a test results (.trx) file...
 No test results files found
```

*_2. *_Right after the above, i have some warnings:

```
WARNING: File is not under the project directory and cannot currently be analysed by SonarQube. File: C:\TFS\Builds\Agent404_01\1\s\Acidizing\BB\Debug\x64\moc\moc_AcidizingDesignerActivity.cpp, project: C:\TFS\Builds\Agent404_01\1\s\Acidizing\AA\TheProject.vcxproj
```

even though the moc_*.cpp files are set to be excluded in Sonar.

**3.**Same for the coverage: 

```
DEBUG - Normalized report includes to '[C:\TFS\Builds\Agent404_01\1\**\*.coveragexml]'
WARN  - Cannot find a report for 'sonar.cxx.coverage.reportPath'
DEBUG - Zeroing coverage information for untouched files
```

**4.** For cpp check, i have issues too:

```
DEBUG - Normalized report includes to '[C:\TFS\Builds\Agent404_01\1\s\Acidizing\.cxxresults\reports-cppcheck\*.xml]'
DEBUG - Adding report 'C:\TFS\Builds\Agent404_01\1\s\Acidizing\.cxxresults\reports-cppcheck\cppcheck-result-0.xml'
INFO  - Processing report 'C:\TFS\Builds\Agent404_01\1\s\Acidizing\.cxxresults\reports-cppcheck\cppcheck-result-0.xml'
INFO  - Parsing 'Cppcheck V2' format
INFO  - Parsing 'Cppckeck V1' format
WARN  - Cannot find the file 'C:\TFS\Builds\Agent404_01\1\s\Acidizing\Engines\BhpEngineInputData.h', skipping violations
WARN  - Cannot find the file 'C:\TFS\Builds\Agent404_01\1\s\Acidizing\Engines\DownholeStageMustBePresentRule.h', skipping violations
```

I thought this runner would run the tests with coverage, that's why i only have the runner active in the configuration. I am trying right now with the following sequence:
1- Run Visual studio build as provided by TFS
2- Run VS Test to generate the coverage and trx
3- Run the msbuild runner to analyze, retrieve results and upload everything to Sonar.

**Comment by jmecosta** [comment]
hi, 

the runner will run the analysis and run static analysis tools cppcheck, rats, vera. so you should see some results when you have rules enabled in your quality profile. At least from your logs i can see cppcheck is correctly imported. Regarding your queries:
1. Coverage, the cxx runner does not run with coverage since there is no generic solution. We use bullseye for example. So in your case you need to create those and use the sonar.cxx.coverage.reportpath=coverage.trx (place this file along with solution file), the wrapper will use absolute  path <SolutionPath>\coverage.trx
2. these are expected, you are trying to import files under output folders Debug which is something not expected
3. see 1
4. that is odd, i would expect those header files to be imported into sonar, can you enable the analysis in debug mode (/g https://github.com/jmecsoftware/sonar-cxx-msbuild-tasks/blob/master/CxxSonarQubeMsbuidRunner/Options.fs#L161) and look in the start of "End Stage" in section of imported files, what are the paths used to lookup import files? If the headers are found in you sonar project but under some other module, then those messages are expected since we use a single cppchek report and parse it to all modules (we will improve this from the cxx plugin side at some time).

To sum up. this is what we do in our organization.

[Pre Step]
1. run intel inspector and produce report-intel.xml -> dump to .cxxresults\other-reports
2. run bullseye and gtest and produce xmls, cov report to .cxxresults\coverage, test reports to .cxxresults\xunit-results

[Analysis]
Build with wrapper. This will take care of the static analysis tools, and vs code analysis and build informaition

**Comment by Jonathan34** [comment]
Ok, funny thing is that to get the tests results, i need to build the tests, and to build the tests, i need to build the solution :). so that s fine, i build, test and then run the wrapper (to build and analyze).

So now it finds the coverage and the .trx.

I think i found out why the results are not uploaded to the server:
`Post-processing succeeded.
 : Cannot Check Analysis results in Server, available only for 5.2 or above` 

But I have sonarqube 5.4...

For 4. it even excludes files that are within the same projet, for instance:
`2016-04-25T19:20:39.6702269Z 14:20:39.513 WARN  - Cannot find the file 'C:\TFS\Builds\Agent404_01\1\s\Common\View\StudyWidget.h', ignoring coverage measures`
and 2 lines below
`2016-04-25T19:20:39.6702269Z 14:20:39.513 DEBUG - Saving '7' coverage measures for file 'C:\TFS\Builds\Agent404_01\1\s\Common\View\GradientViewModel.cpp'`
i will run with the /g flag as soon as the issue above is fixed.

Also alternate problem that is probably linked to Cxx (you will confirm):
If i do the clean build/tests/analysis in debug, I end up with:

```
DEBUG - Saving '7' coverage measures for file 'C:\TFS\Builds\Agent404_01\1\s\Acidizing\View\AcidizingCoiledTubingView.cpp'
ERROR - Saving cov measure 'coverage_line_hits_data' = 'null' failed
ERROR - Obtained 'coverage_line_hits_data' = '8=0;9=0;10=0;11=0;14=0;15=0;16=0;31=0' failed
ERROR - Ctype : 'UT_COVERAGE'     Exception 'Can't create measure for line 31 for file 'C:/TFS/Builds/Agent404_01/1/s/Acidizing/View/AcidizingCoiledTubingView.cpp' with 17 lines'
 Caused by: java.lang.IllegalStateException: Can't create measure for line 31 for file 'C:/TFS/Builds/Agent404_01/1/s/Acidizing/View/AcidizingCoiledTubingView.cpp' with 17 lines
```

In release, this error doe snto appear.

**Comment by jmecosta** [comment]
ok, now i see what is your problem. and as far as i know sonarqube does not support this kind of setup.

the problem is that you solution and sources need to be in same folder or in subfolders of solution. in your case you have the solution in

C:\TFS\Builds\Agent404_01\Acidizing -> this is your base path

and you have sources in
C:\TFS\Builds\Agent404_01\1 

everything outside folder Acidizing will not be imported. Its not so much a problem of my wrapper since i use absolute paths so everything should be looked up properly. the issue is just that the platform does not support this.

Regarding the 

```
Cannot Check Analysis results in Server, available only for 5.2 or above
```

we also use 5.4 and we have this in our logs:

```
[00:40:37] :         [Step 3/3] 00:40:37.277 INFO  - Analysis reports compressed in 1067ms, zip size=1 MB
[00:40:37] :         [Step 3/3] 00:40:37.277 DEBUG - Upload report
[00:40:37] :         [Step 3/3] 00:40:37.356 DEBUG - POST 200 http://sonar/api/ce/submit?projectKey=Project:Name&projectName=Name | time=79ms
[00:40:37] :         [Step 3/3] 00:40:37.366 INFO  - Analysis report uploaded in 89ms
[00:40:37] :         [Step 3/3] 00:40:37.366 INFO  - ANALYSIS SUCCESSFUL, you can browse http://sonar/dashboard/index/Project:Name
[00:40:37] :         [Step 3/3] 00:40:37.367 INFO  - Note that you will be able to access the updated dashboard once the server has processed the submitted analysis report
[00:40:37] :         [Step 3/3] 00:40:37.367 INFO  - More about the report processing at http://sonar/api/ce/task?id=AVRPXfbU2A0lc9RFi6zA
[00:40:37] :         [Step 3/3] 00:40:37.375 DEBUG - Report metadata written to E:\prod\structures\src\work\Core\Name\.sonarqube\out\.sonar\report-task.txt
[00:40:37] :         [Step 3/3] 00:40:37.376 DEBUG - Post-jobs : 
[00:40:37] :         [Step 3/3] 00:40:37.376 INFO  - Analysis report generated in E:\prod\structures\src\work\Core\Name\.sonarqube\out\.sonar\batch-report
[00:40:37] :         [Step 3/3] INFO: ------------------------------------------------------------------------
[00:40:37] :         [Step 3/3] INFO: EXECUTION SUCCESS
[00:40:37] :         [Step 3/3] INFO: ------------------------------------------------------------------------
[00:40:37] :         [Step 3/3] Total time: 6:56.111s
[00:40:38] :         [Step 3/3] Final Memory: 216M/739M
[00:40:38] :         [Step 3/3] INFO: ------------------------------------------------------------------------
[00:40:38] :         [Step 3/3] Process returned exit code 0
[00:40:38] :         [Step 3/3] The SonarQube Scanner has finished
[00:40:38] :         [Step 3/3] 00:40:38.766  Creating a summary markdown file...
[00:40:38] :         [Step 3/3] 00:40:38.767  Analysis results: http://sonar/dashboard/index/Project:Name
[00:40:38] :         [Step 3/3] Post-processing succeeded.
[00:40:38] :         [Step 3/3] [EndPhase] : Check Analysis Results in Server every 2 seconds
[00:40:38] :         [Step 3/3] [SERVER CALL] http://sonar/api/ce/task?id=AVRPXfbU2A0lc9RFi6zA
[00:40:38] :         [Step 3/3] STATUS: PENDING 
[00:40:39] :         [Step 3/3] [SERVER CALL] http://sonar/api/ce/task?id=AVRPXfbU2A0lc9RFi6zA
[00:40:39] :         [Step 3/3] STATUS: IN_PROGRESS 
[00:40:40] :         [Step 3/3] [SERVER CALL] http://sonar/api/ce/task?id=AVRPXfbU2A0lc9RFi6zA
[00:40:40] :         [Step 3/3] STATUS: IN_PROGRESS 
[00:40:41] :         [Step 3/3] [SERVER CALL] http://sonar/api/ce/task?id=AVRPXfbU2A0lc9RFi6zA
[00:40:41] :         [Step 3/3] STATUS: IN_PROGRESS 
[00:40:42] :         [Step 3/3] [SERVER CALL] http://sonar/api/ce/task?id=AVRPXfbU2A0lc9RFi6zA
[00:40:42] :         [Step 3/3] STATUS: IN_PROGRESS 
[00:40:43] :         [Step 3/3] [SERVER CALL] http://sonar/api/ce/task?id=AVRPXfbU2A0lc9RFi6zA
[00:40:44] :         [Step 3/3] STATUS: IN_PROGRESS 
```

in your log you need to see: 

```
[00:40:37] :         [Step 3/3] 00:40:37.367 INFO  - More about the report processing at http://sonar/api/ce/task?id=AVRPXfbU2A0lc9RFi6zA
```

this will tell me that you are running 5.2 or above, if that is not there then seems you are not running 5.4

**Comment by Jonathan34** [comment]
Interesting to see it does not support a solution in a subfolder of the root... Looks like i could disregard it using your wrapper though... worth a try.

The last lines of the log are as follow, i do not have EXECUTION SUCCESS etc

```
2016-04-25T19:20:41.5921759Z 14:20:41.592 DEBUG - Populating index from [moduleKey=CORAL:CORAL:FD24815A-8149-4B8A-9945-A5A03BF416C6, relative=precompiled.h, basedir=C:\TFS\Builds\Agent404_01\1\s\Common\View]
2016-04-25T19:20:41.5921759Z 14:20:41.592 INFO  - Sensor CPD Sensor (wrapped) (done) | time=578ms
2016-04-25T19:20:41.5921759Z 14:20:41.592 INFO  - -------------  Scan Coral
2016-04-25T19:20:41.6390538Z 14:20:41.639 INFO  - Excluded sources for coverage: 
2016-04-25T19:20:41.6390538Z 14:20:41.639 INFO  -   *\\**\\ui_*.cpp
2016-04-25T19:20:41.6390538Z 14:20:41.639 INFO  -   *\\**\\moc_*.cpp
2016-04-25T19:20:41.6546787Z 14:20:41.639 INFO  -   *\\**\\GeneratedFiles\\*
2016-04-25T19:20:41.6546787Z 14:20:41.639 INFO  -   *\\**\\TL_build\\*
2016-04-25T19:20:41.6546787Z 14:20:41.639 INFO  -   *\\**\\3rdParty\\*
2016-04-25T19:20:41.6546787Z 14:20:41.639 DEBUG - Initializers : 
2016-04-25T19:20:41.6546787Z 14:20:41.639 INFO  - Base dir: C:\TFS\Builds\Agent404_01\1\s
2016-04-25T19:20:41.6546787Z 14:20:41.639 INFO  - Working dir: C:\TFS\Builds\Agent404_01\1\.sonarqube\out\.sonar
2016-04-25T19:20:41.6546787Z 14:20:41.639 INFO  - Source encoding: UTF-8, default locale: en_US
2016-04-25T19:20:41.7484324Z 14:20:41.748 INFO  - JaCoCoSensor: JaCoCo report not found : C:\TFS\Builds\Agent404_01\1\s\target\jacoco.exec
2016-04-25T19:20:41.7640576Z 14:20:41.748 INFO  - JaCoCoItSensor: JaCoCo IT report not found: C:\TFS\Builds\Agent404_01\1\s\target\jacoco-it.exec
2016-04-25T19:20:41.7640576Z 14:20:41.748 DEBUG - Sensors : Lines Sensor (wrapped) -> org.sonar.plugins.cxx.tests.dotnet.CxxUnitTestResultsProvider$CxxUnitTestResultsImportSensor@1acec5e -> SCM Sensor (wrapped) -> Code Colorizer Sensor (wrapped) -> CPD Sensor (wrapped)
2016-04-25T19:20:41.7640576Z 14:20:41.748 INFO  - Sensor Lines Sensor (wrapped)
2016-04-25T19:20:41.7640576Z 14:20:41.748 INFO  - Sensor Lines Sensor (wrapped) (done) | time=0ms
2016-04-25T19:20:41.7640576Z 14:20:41.748 INFO  - Sensor org.sonar.plugins.cxx.tests.dotnet.CxxUnitTestResultsProvider$CxxUnitTestResultsImportSensor
2016-04-25T19:20:42.7640954Z Process returned exit code 0
2016-04-25T19:20:42.7640954Z The SonarQube Scanner has finished
2016-04-25T19:20:42.7797221Z 14:20:42.764  Creating a summary markdown file...
2016-04-25T19:20:42.7797221Z 14:20:42.779  Analysis results: http://us1153app47.dir.slb.com:8888/dashboard/index/CORAL
2016-04-25T19:20:42.7953475Z Post-processing succeeded.
2016-04-25T19:20:43.1234851Z [EndPhase] : Cannot Check Analysis results in Server, available only for 5.2 or above
```

Here is a screenshot of the server system info: http://screenpresso.com/=6T4lf 
It is really SQ 5.4 :)

**Comment by jmecosta** [comment]
thats all of the log? i mean nothing from the cxx plugin is running. this is a typical line of a run:

```
[00:33:48] :         [Step 3/3] 00:33:48.243 DEBUG - Code colorizer, supported languages: cs,c++,ail,xml,msbuild,py,fs
[00:33:48] :         [Step 3/3] 00:33:48.264 DEBUG - Sensors : Lines Sensor (wrapped) -> QProfileSensor -> CxxSquidSensor -> CxxRatsSensor -> CxxCoverageSensor -> CxxCppCheckSensor -> CxxCompilerSensor -> CxxVeraxxSensor -> CxxExternalRulesSensor -> MSBuildLineCounterSensor -> org.sonar.plugins.msbuild.projectchecker.MSBuildProjectCheckerExtensionSensor@6839203b -> org.sonar.plugins.msbuild.dgmlcreator.MSBuildDiagramCreatorSensor@2d28fb02 -> org.sonar.plugins.csharp.CSharpCodeCoverageProvider$CSharpCoverageReportImportSensor@6db328f8 -> org.sonar.plugins.csharp.CSharpUnitTestResultsProvider$CSharpUnitTestResultsImportSensor@402f8592 -> org.sonar.plugins.fsharp.FSharpCodeCoverageProvider$FSharpCoverageReportImportSensor@61b65d54 -> org.sonar.plugins.fsharp.FSharpUnitTestResultsProvider$FSharpUnitTestResultsImportSensor@b2c4a8b -> SCM Sensor (wrapped) -> Code Colorizer Sensor (wrapped) -> CPD Sensor (wrapped)
[00:33:48] :         [Step 3/3] 00:33:48.284 ERROR - License for views is expired
[00:33:48] :         [Step 3/3] 00:33:48.284 INFO  - Sensor Lines Sensor (wrapped)
[00:33:48] :         [Step 3/3] 00:33:48.363 INFO  - Sensor Lines Sensor (wrapped) (
```

As you can see i have many more sensors, now the problem is due to problem of the subfolders and that explains also why you dont have any of these:

```
[00:33:46] :         [Step 3/3] 00:33:46.630 DEBUG - Declared extensions of language C# were converted to sonar.lang.patterns.cs : **/*.cs
[00:33:46] :         [Step 3/3] 00:33:46.630 DEBUG - Declared extensions of language Java were converted to sonar.lang.patterns.java : **/*.java,**/*.jav
[00:33:46] :         [Step 3/3] 00:33:46.630 DEBUG - Declared extensions of language XML were converted to sonar.lang.patterns.xml : **/*.xml
[00:33:46] :         [Step 3/3] 00:33:46.635 DEBUG - Declared extensions of language F# were converted to sonar.lang.patterns.fs : **/*.fs,**/*.fsx,**/*.fsi
[00:33:46] :         [Step 3/3] 00:33:46.776 DEBUG - Language of file 'includes.props' is detected to be 'msbuild'
```

or you are using some weird versions of the plugins in server. can you share what versions of the C# and C++ plugins are u using in server?

**Comment by Jonathan34** [comment]
The plugin info: http://screenpresso.com/=EYMrc
I sent you a full log file by email. Github does not let me upload a zip file...

I do have all the sensors and the declared languages as you do.

**Comment by jmecosta** [comment]
sure, please send to jmecosta@gmail.com
